### PR TITLE
refactor(api): create robot-server subproject

### LINKED ISSUE #5011: Create api-server sub-project
## overview
This is the api server application. For the time being both the aiohttp (opentrons.server) and fastapi (opentrons.app) packages will be in this project.
Directory Structure
```
api
| README.md
| Makefile
| ...
└- tests
|  | __init__.py
|  └- app
└- app
   | __init__.py
   | main.py
   └- aiohttp ( to be deprecated )
   └- fastapi ( move contents one level up when aiohttp is deprecated )
```

- [x] directory structure is created
- [x] create pipfile 
- [ ] move webservice related documentation from `api` to new `api-server`

### PR DESCRIPTION
**Pull request by amitlissack**
## overview

api is being split up into two projects. This creates the webservice portion of the split.

The whole `api` project was copied to a new monorepo sub-project called `robot-server`.
The server (aiohttp) and app (fastapi) packages were copied over along with the tests. Everything else was deleted.

## review requests

This is a work in progress.  Notice that it's to be merged into `api_splitup`, a feature branch for the split up.

closes #5011

### CODE REVIEW

File: robot-server/Makefile [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+docs/dist/ot1: docs/ot1
+	$(SHX) mkdir -p $@
+	$(SHX) cp -R $< $(@D)
+
+docs/dist/%: docs/root/%
+	$(SHX) mkdir -p $(@D)
+	$(SHX) cp -R docs/root/* docs/dist/
+
+.PHONY: docs
+docs: docs/dist/index.html docs/dist/v1 docs/dist/v2 docs/dist/ot1
```
Comment (mcous) [inline robot-server/Makefile]: I'm not sure all the docs stuff belongs in `robot-server`. This documentation is all for the <https://docs.opentrons.com> site, which details how to use the Python Protocol API (which still lives in `api`)

File: robot-server/README.rst [hunk 2]
```diff
@@ -0,0 +1,114 @@
+=============
+Opentrons API
```
Comment (mcous) [inline robot-server/README.rst]: Is this file a copy paste? Looks like it's not quite accurate to (my understanding of) how this split is currently working.

We should also be careful about the term "Opentrons API", which in our users' minds refers to the Python Protocol API. IMO this project should the "Opentrons OT-2 HTTP API" or something similar

File: robot-server/Makefile (hunk 1 above)
Comment (amitlissack) [reply robot-server/Makefile]: There's a separate ticket for Makefile related issues on this new app. I'll kick this until then.

File: robot-server/README.rst (hunk 2 above)
Comment (amitlissack) [reply robot-server/README.rst]: Good call. I hadn't touched the docs.

File: robot-server/Makefile (hunk 1 above)
Comment (mcous) [reply robot-server/Makefile]: Besides the Makefile stuff there's a _bunch_ of files in `docs` for this PR that were (correctly) not deleted from `api` as part of #5052. They should probably be removed from this PR so as to not accidentally duplicate them.

Just looking at the diff, #5052 was -10000, so it's weird that this PR is +70000

Comment (amitlissack) [reply robot-server/Makefile]: Okeedoke

File: robot-server/Makefile [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+	-$(python) -m twine upload --repository-url "https://test.pypi.org/legacy/"\
+                             $(twine_auth_args)\
+                             $(wheel_file)
+
+.PHONY: deploy
+deploy: wheel
+	$(python) -m twine upload $(twine_auth_args)\
+                            $(wheel_file)
+
+.PHONY: push-balena
```
Comment (sfoster1) [inline robot-server/Makefile]: I think we can remove this, balena's been gone for a while now

File: robot-server/Makefile [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+	curl -X POST \
+		-H "Content-Type: multipart/form-data" \
+		-F "whl=@$(wildcard dist/opentrons-*.whl)" \
+		http://$(host):31950/server/update
+
+.PHONY: push
+push: wheel
+	scp -i $(br_ssh_key) $(ssh_opts) $(wheel_file) root@$(host):/data/
+	ssh -i $(br_ssh_key) $(ssh_opts) root@$(host) \
+"function cleanup () { rm -f /data/opentrons*.whl && mount -o remount,ro / && systemctl start opentrons-api-server; } ;\
```
Comment (sfoster1) [inline robot-server/Makefile]: This probably needs to change - the wheel for this project doesn't look like this, and it's the wrong systemd unit

File: robot-server/Makefile [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+systemctl restart jupyter-notebook &&\
```
Comment (sfoster1) [inline robot-server/Makefile]: shouldn't need this anymore I think

File: robot-server/Makefile [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+dev: export ENABLE_VIRTUAL_SMOOTHIE := true
+dev:
+	$(python) -m robot_server.main -P 31950
+
+.PHONY: local-shell
+local-shell:
+	$(pipenv_envvars) pipenv shell
+
+.PHONY: simulate
+simulate:
```
Comment (sfoster1) [inline robot-server/Makefile]: doesn't really apply to this project I think

File: robot-server/Makefile [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+pypi_password ?=
+
+# Host key location for buildroot robot
+br_ssh_key ?= ~/.ssh/robot_key
+# Pubkey location for buildroot robot to install with install-key
+br_ssh_pubkey ?= $(br_ssh_key).pub
+# Other SSH args for buildroot robots
+ssh_opts ?= -o stricthostkeychecking=no -o userknownhostsfile=/dev/null
+
+twine_auth_args := --username $(pypi_username) --password $(pypi_password)
```
Comment (sfoster1) [inline robot-server/Makefile]: we're not putting this on pypi so I don't think we need twine stuff

File: robot-server/Makefile [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+# These variables can be overriden when make is invoked to customize the
+# behavior of pytest. For instance,
+# make test tests=tests/opentrons/tools/test_qc_scripts.py would run only the
+# specified test
+tests ?= tests/robot_server
+test_opts ?=  --cov=$(SRC_PATH) --cov-report term-missing:skip-covered --cov-report xml:coverage.xml
+
+# These variables must be overridden when make deploy or make deploy-staging is run
+# to set the auth details for pypi
+pypi_username ?=
```
Comment (sfoster1) [inline robot-server/Makefile]: don't need pypi stuff

File: robot-server/Makefile [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+# and also make an explicit version for shx for use in the shell function,
+# where PATH won’t be propagated
+PATH := $(shell cd .. && yarn bin):$(PATH)
+SHX := npx shx
+
+# Path of source package
+SRC_PATH = robot_server
+
+# make push wheel file (= rather than := to expand at every use)
+firmware = $(wildcard smoothie/*.hex)
```
Comment (sfoster1) [inline robot-server/Makefile]: don't think we need this in the http side

File: robot-server/Pipfile [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+pytest-cov = "*"
+mypy = "==0.740"
+atomicwrites = {version="*", sys_platform="== 'win32'"}
+
+[packages]
+uvicorn = "==0.11.3"
+fastapi = "==0.49.0"
+python-multipart = "==0.0.5"
+aiohttp = "==3.4.4"
+jsonrpcserver = "*"
```
Comment (sfoster1) [inline robot-server/Pipfile]: I think this is a hardware controller concern and doesn't need to be here

File: robot-server/buildroot.mk [hunk 11]
```diff
@@ -0,0 +1,49 @@
+################################################################################
+#
+# python-opentrons-api
+#
+################################################################################
+
+# Get a key from package.json (like version)
+define get_pkg_json_key
+	$(shell python -c "import json; print(json.load(open('$(BR2_EXTERNAL_OPENTRONS_MONOREPO_PATH)/api/src/opentrons/package.json'))[\"$(1)\"])")
```
Comment (sfoster1) [inline robot-server/buildroot.mk]: this path needs to be updated (or this funciton should be changed, fine with that too)

File: robot-server/buildroot.mk [hunk 12]
```diff
@@ -0,0 +1,49 @@
+################################################################################
+#
+# python-opentrons-api
```
Comment (sfoster1) [inline robot-server/buildroot.mk]: I think the package name should be `python-opentrons-robot-server` or something

File: robot-server/opentrons-api-server.service [hunk 13]
```diff
@@ -0,0 +1,14 @@
+[Unit]
+Description=Opentrons API server
+Requires=nginx.service
+After=nginx.service
+
+[Service]
+Type=exec
+ExecStart=python -m opentrons.main -U /run/aiohttp.sock --hardware-server
```
Comment (sfoster1) [inline robot-server/opentrons-api-server.service]: This stuff needs to be changed I think

File: robot-server/pypi-readme.rst [hunk 14]
```diff
@@ -0,0 +1,51 @@
+.. _Full API Documentation: http://docs.opentrons.com
```
Comment (sfoster1) [inline robot-server/pypi-readme.rst]: I don't believe we need this because it's all for pypi

File: robot-server/pytest.ini [hunk 15]
```diff
@@ -0,0 +1,8 @@
+[pytest]
+markers =
+        api1_only: Test only functions using API version 1 (legacy_api)
```
Comment (sfoster1) [inline robot-server/pytest.ini]: may not need these markers anymore

File: robot-server/release-notes.md [hunk 16]
```diff
@@ -0,0 +1,54 @@
+# Robot OS Changes from 3.14.1 to 3.15.2
```
Comment (sfoster1) [inline robot-server/release-notes.md]: I think this should live in the api  still? open to discussion though

**Comment by amitlissack** [comment]
@sfoster1 The goal is to remove the aiohttp package, isn't it? Is the history so important? I can redo as a `git mv`. I'd really rather not keep one foot in and one foot out. (the feet are metaphors for `api` and `robot-server`... bleh).

Also, I should have mentioned this, but the build and Makefile changes will come in a separate ticket/PR. Your comments will not be ignored. Is that cool?

**Comment by amitlissack** [comment]
Abandoning this PR in favor of #5083

**Pull request closed without merging**